enterprise_access/apps/content_assignments/tasks.py

```diff
@@ -131,7 +131,7 @@ def send_cancel_email_for_pending_assignment(cancelled_assignment_uuid):
         return
     except Exception as exc:  # pylint: disable=broad-exception-caught
         logger.error(f"Unable to send email for {lms_user_id} due to exception: {exc}")
-        learner_content_assignment_action.error_reason = AssignmentActionErrors.EMAIL_ERROR
+        learner_content_assignment_action.error_reason = AssignmentActionErrors.CANCEL_ERROR
```

Comment:
[question]: curious if it would be a better approach to keep the two existing error reasons as is but add an action type.

```
learner_content_assignment_action.error_reason = { 
   error_reason: AssignmentActionErrors.CANCEL_ERROR,
   action_type: AssignmentActions.CANCELLED,
}
```
so the data could look something like this similar to what Adam suggested [here] (https://github.com/openedx/frontend-app-admin-portal/pull/1099/files#r1400722639) and we wouldn't need to add additional error types. 
```
"error_reason":  { 
   "action_type": "cancelled", 
   "error_reason": "internal_api_error" 
}
```

Reply:
FYI, the `LearnerContentAssignmentAction.action_type` field already exists, and `cancelled` is one of the choices.